File: output/codeium/python/easy/q6-valid-anagram.py
# https://leetcode.com/problems/valid-anagram/
# Problem: Valid Anagram

#Problem Description:
#Given two strings s and t, return true if t is an anagram of s, and false otherwise.

# Constraints:
# 1 <= s.length, t.length <= 5 * 10^4
# s and t consist of lowercase English letters.

#Code Structure:
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """    
        if len(s) != len(t):
            return False
        sCount = {}
        tCount = {}
        for c1, c2 in zip(s, t):
            sCount[c1] = sCount.get(c1, 0) + 1
            tCount[c2] = tCount.get(c2, 0) + 1
        for c in sCount:
            if sCount[c] != tCount.get(c, 0):
                return False
        return True
    

File: output/sonarqube_analysis/codeium_solutions/javascript/results/q6-valid-anagram.js
/** https://leetcode.com/problems/valid-anagram */
// Problem: Valid Anagram

//Problem Description:
//Given two strings s and t, return true if t is an anagram of s, and false otherwise.

// Constraints:
// 1 <= s.length, t.length <= 5 * 10^4
// s and t consist of lowercase English letters.

//Code Structure:
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) return false;
    const sCount = {};
    const tCount = {};
    for (let i = 0; i < s.length; i++) {
        sCount[s[i]] = (sCount[s[i]] || 0) + 1;
        tCount[t[i]] = (tCount[t[i]] || 0) + 1;
    }
    for (const c in sCount) if (sCount[c] !== tCount[c]) return false;
    return true;
};
    